Assets/Scripts/Weapons/BehaviourTypes/BulletMono.cs

```diff
@@ -2,50 +2,31 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-//To be attached to Bullet Objects
+// Behaviour for Bullet game objects
 public class BulletMono : MonoBehaviour
 {
-    float _damage = 10f;
+    private BulletAttackBehaviour _attackBehaviour;
 
-    float _bulletSpeed = 10f;
-
-    float _bulletSpawnTime;
-
-    EntityType _owner;
-
-    public static void create(Vector3 position, Vector3 direction, Dictionary<string, float> statsModifiers, EntityType owner)
+    public static void create(BulletAttackBehaviour attackBehaviour, Vector3 position, Vector3 direction)
     {
-        GameObject newObject = Instantiate(Resources.Load("Bullet")) as GameObject;
-        newObject.transform.position = position;
-        newObject.transform.forward = direction;
+        GameObject newBullet = Instantiate(Resources.Load("Bullet")) as GameObject;
 
-        BulletMono bulletComponent = newObject.GetComponent<BulletMono>();
-        bulletComponent._damage += statsModifiers["damage"];
-        bulletComponent._bulletSpeed += statsModifiers["bulletSpeed"];
-        bulletComponent._owner = owner;
+        newBullet.GetComponent<BulletMono>()._attackBehaviour = attackBehaviour;
+        
+        newBullet.transform.position = position;
+        newBullet.transform.forward = direction;
     }
 
     void Update()
     {
-        transform.position += transform.forward * _bulletSpeed * Time.deltaTime * SpeedManager.bulletSpeedScaling;
+        transform.position += transform.forward * _attackBehaviour._bulletSpeed * Time.deltaTime * SpeedManager.bulletSpeedScaling;
     }
 
-    void OnTriggerEnter(Collider c){
-        Stats statsComponent = c.gameObject.GetComponent<Stats>();
-        if (statsComponent){
-            if (statsComponent.owner != _owner){
-                statsComponent.takeDamage(_damage);
-                Destroy(gameObject); // bullet
-            }
-        } else {
+    void OnTriggerEnter(Collider c)
+    {
+        if (!_attackBehaviour.onHit(c.gameObject))
```

Comment:
I feel like we should rename `onHit` to something else, it returning `True` or `False` isn't really intuitive to me. Unless I'm missing something.
Or maybe flip the boolean so it means if it `hit` something it should destroy itself

Reply:
I thought about this as well, and I agree it's unintuitive to have onHit return a boolean for whether or not the attack should continue. It's not particularly applicable to all attacks. I'm going to change this shortly.